database/seeders/DatabaseSeeder.php

```diff
@@ -2,19 +2,51 @@
 
 namespace Database\Seeders;
 
-// use Illuminate\Database\Console\Seeds\WithoutModelEvents;
+use App\Models\Company;
+use App\Models\Song;
 use Database\Seeders\CompanySeeder;
+use Database\Seeders\RequestedSongSeeder;
+use Database\Seeders\SongSeeder;
+use Database\Seeders\UserSeeder;
 use Illuminate\Database\Seeder;
 
 class DatabaseSeeder extends Seeder
 {
-  /**
-   * Seed the application's database.
-   */
   public function run(): void
   {
     $this->call([
+      SongSeeder::class,
       CompanySeeder::class,
+      UserSeeder::class,
     ]);
+
+    $this->seedRandomSongsByCompany();
+    $this->seedBlacklistedSongs();
+
+    $this->call([
+      RequestedSongSeeder::class,
+      UpvoteSeeder::class
+    ]);
+  }
+
+  protected function seedRandomSongsByCompany()
+  {
+    $songs = Song::all()->take(5);
+    $companies = Company::all();
+    foreach ($songs as $song) {
+      $randomCompany = $companies->random();
+      $randomCompany->songs()->attach($song->id);
+    }
+  }
+
+  protected function seedBlacklistedSongs()
+  {
+    $blacklistedSongs = collect(Song::all())->slice(4, 2);
```

Comment:
remove collect()

Reply:
Also, you should pick blacklisted song in random order

Reply:
Integrated a system when we take 5 songs using `limit `to optimize a query  ` $blacklistedSongs = Song::limit(5)->get()`; Then I distribute those songs randomly among companies